spec/ameba/rule/lint/initialize_definition_with_receiver_spec.cr

```diff
@@ -0,0 +1,39 @@
+require "../../../spec_helper"
+
+module Ameba::Rule::Lint
+  describe InitializeDefinitionWithReceiver do
+    subject = InitializeDefinitionWithReceiver.new
+
+    {% for keyword in %w[struct class module].map(&.id) %}
+      it "passes for `initialize` method definition without a receiver" do
+        expect_no_issues subject, <<-CRYSTAL
+          {{ keyword }} Foo
+            def initialize
+            end
+          end
+          CRYSTAL
+      end
+
+      it "fails for `initialize` method definition with a `self` receiver" do
+        expect_issue subject, <<-CRYSTAL
+          {{ keyword }} Foo
+            def self.initialize
+          # ^^^^^^^^^^^^^^^^^^^ error: `initialize` method definition should not have a receiver
+            end
+          end
+          CRYSTAL
+      end
+
+      it "fails for `initialize` method definition with a receiver" do
+        expect_issue subject, <<-CRYSTAL
+          {{ keyword }} Foo
+          end
+
+          def Foo.initialize
```

Comment:
I'm not sure if this case should be flagged since it's more likely intentional.

Reply:
is there a difference between `def Foo.initialize` and just `def initialize` ?